### ISSUE TITLE
fail to install on amazonlinux 2

### ISSUE THREAD

**Issue by fams** [christiangda/ansible-role-amazon-cloudwatch-agent#22]
Im getting an error trying to use this role in amazon linux 2
```
The conditional check 'cwa_conf_json_file_content is search("collectd")' failed. The error was: Unexpected templating type error occurred on ({% if cwa_conf_json_file_content is search("collectd") %} True {% else %} False {% endif %}): expected string or buffer

The error appears to be in '/root/.ansible/roles/christiangda.amazon_cloudwatch_agent/tasks/install-redhat.yml': line 45, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  block:
    - name: Install "{{ cwa_dependencies_packages }}" dependency on Amazon 1/2
      ^ here
We could be wrong, but this one looks like it might be an issue with
missing quotes. Always quote template expression brackets when they
start a value. For instance:

    with_items:
      - {{ foo }}

Should be written as:

    with_items:
      - "{{ foo }}"
```
My config is 
```ansible
    - role: christiangda.amazon_cloudwatch_agent
      vars:
        cwa_agent_mode: "ec2"
        cwa_conf_json_file_content: "{{ lookup('file', 'files/CloudWatch.json') | from_json }}"
````
contents 
```json
{
    "agent": {
        "metrics_collection_interval": 60,
        "region": "eu-west-1",
        "logfile": "/opt/aws/amazon-cloudwatch-agent/logs/amazon-cloudwatch-agent.log",
        "debug": false
    },
    "logs": {
        "logs_collected": {
            "files": {
                "collect_list": [
                    {
                        "file_path": "/var/log/message",
                        "log_group_name": "/var/log/message",
                        "log_stream_name": "{instance_id}"
                    }
                ]
            }
        }
    },
    "metrics": {
        "append_dimensions": {
            "AutoScalingGroupName": "${aws:AutoScalingGroupName}",
            "ImageId": "${aws:ImageId}",
            "InstanceId": "${aws:InstanceId}",
            "InstanceType": "${aws:InstanceType}"
        },
        "metrics_collected": {
            "cpu": {
                "measurement": [
                    "cpu_usage_idle",
                    "cpu_usage_iowait",
                    "cpu_usage_user",
                    "cpu_usage_system"
                ],
                "metrics_collection_interval": 60,
                "resources": [
                    "*"
                ],
                "totalcpu": false
            },
            "disk": {
                "measurement": [
                    "used_percent",
                    "inodes_free"
                ],
                "metrics_collection_interval": 60,
                "resources": [
                    "*"
                ]
            },
            "diskio": {
                "measurement": [
                    "io_time",
                    "write_bytes",
                    "read_bytes",
                    "writes",
                    "reads"
                ],
                "metrics_collection_interval": 60,
                "resources": [
                    "*"
                ]
            },
            "mem": {
                "measurement": [
                    "mem_used_percent"
                ],
                "metrics_collection_interval": 60
            },
            "netstat": {
                "measurement": [
                    "tcp_established",
                    "tcp_time_wait"
                ],
                "metrics_collection_interval": 60
            },
            "swap": {
                "measurement": [
                    "swap_used_percent"
                ],
                "metrics_collection_interval": 60
            }
        }
    }
}
```

**Comment by mishawakerman** [comment]
I was having the same issue and have a workaround.

## The Problem

I looks like `search` expects `cwa_conf_json_file_content` to be a `string` or `buffer` but it is of type `dict`:

```
---
- hosts: servers
  gather_facts: True
  vars:
    cwa_conf_json_file_content: "{{ lookup('file', 'cloudwatch/api.json') | from_json }}"
  tasks:
    - debug:
        msg: '{{ cwa_conf_json_file_content | type_debug }}'
```
```
TASK [debug] ************************************************************************************************************************************************************************
ok: [example-pilot-api] => {
    "msg": "dict"
}
```

## The Workaround
I was having the same issue and fixed it as follows:

**Before `.ansible/roles/christiangda.amazon_cloudwatch_agent/tasks/install-debian.yml`**
```
...
- name: Install dependencies on Debian Family
  apt:
    name: "{{ cwa_dependencies_packages }}"
    state: present
    force: true
  register: package_ok
  retries: "{{ cwa_global_downloads_retries }}"
  delay: "{{ cwa_global_downloads_delay }}"
  until: package_ok is success
  when: cwa_conf_json_file_content is search("collectd")
  tags:
    - install
...
```

**After `.ansible/roles/christiangda.amazon_cloudwatch_agent/tasks/install-debian.yml`**
```
...
- name: Install dependencies on Debian Family
  apt:
    name: "{{ cwa_dependencies_packages }}"
    state: present
    force: true
  register: package_ok
  retries: "{{ cwa_global_downloads_retries }}"
  delay: "{{ cwa_global_downloads_delay }}"
  until: package_ok is success
  when: "'collectd' in cwa_conf_json_file_content"    # <- different
  tags:
    - install
...
```

I'm not sure if this is an ansible version issue or something else. @christiangda - let me know if you'd like a PR.

**Comment by christiangda** [comment]
Hi @fams ,

This looks great! thank you very much! could you make a pull request using the `develop` branch please, to see if the tests run fine?.

The other thing is, could you apply the same changes for RedHat and AWS family if those exist?

**Comment by Jno21** [comment]
I made a PR with @mishawakerman solution. It needs to be tested.
When this is merge can we have a new version coming up and published on ansible-galaxy ? :)

This error is totally blocking my use case currently.

**Comment by Jno21** [comment]
Hi ! I tested it yesterday on debian. Seems to work fine, the error does not show up again.

**Comment by christiangda** [comment]
Fixed in release 2.0.5
